test/src/test/java/org/corfudb/runtime/view/StreamViewTest.java

```diff
@@ -382,4 +382,31 @@ public void prefixTrimThrowsException()
         assertThatThrownBy(() -> sv.next())
                 .isInstanceOf(TrimmedException.class);
     }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void trimCausesCacheEviction()
+            throws Exception {
+        UUID streamA = UUID.nameUUIDFromBytes("stream A".getBytes());
+        UUID streamB = UUID.nameUUIDFromBytes("stream B".getBytes());
+        byte[] testPayload = "hello world".getBytes();
+
+        IStreamView svA = r.getStreamsView().get(streamA);
+        IStreamView svB = r.getStreamsView().get(streamB);
+
+        svA.append(testPayload);
+        svB.append(testPayload);
+
+        getRuntime().getAddressSpaceView().prefixTrim(1);
+        getRuntime().getAddressSpaceView().invalidateClientCache();
+        getRuntime().getAddressSpaceView().invalidateServerCaches();
+
+        // First witness of TrimmedException by AddressSpaceView::read()
+        assertThatThrownBy(() -> svB.next().getPayload(getRuntime()))
+                .isInstanceOf(TrimmedException.class);
+        // Test short-cut in AddressSpaceView::read() by trying
+        // to read an earlier log address than svB's attempt.
+        assertThatThrownBy(() -> svA.next().getPayload(getRuntime()))
+                .isInstanceOf(TrimmedException.class);
```

Comment:
A's append is at address 0, B's append is at address 1.  This line is A's first attempt to read the stream, so the trim at 1 should trigger TrimmedException.